src/main/java/org/maxgamer/rs/cache/Cache.java

```diff
@@ -14,8 +14,77 @@
 
 import org.maxgamer.rs.cache.reference.Reference;
 import org.maxgamer.rs.cache.reference.ReferenceTable;
+import org.maxgamer.rs.structure.configs.FileConfig;
+import org.maxgamer.rs.util.Log;
 
 public class Cache {
+	/**
+	 * Initializes a new cache. This removes any encrypted files from the cache which can't be read
+	 * @return a newly initialized cache
+     */
+	public static Cache init() throws IOException {
+        Log.debug("Loading Cache...");
+        Cache cache = new Cache();
+        cache.load(new File("cache"));
+
+        //We store a file as data/cache.yml, this file contains information on files which we should
+        //delete from our cache (Encrypted maps). This is done to avoid sending the player maps that
+        //we do not have an XTEA key for.
+        FileConfig cacheCfg = new FileConfig(new File("data", "cache.yml"));
+        cacheCfg.reload();
+        //The reference table as a ByteBuffer.
+        CacheFile f = cache.getFile(255, IDX.LANDSCAPES);
+        //The decoded reference table
+        ReferenceTable r = cache.getReferenceTable(IDX.LANDSCAPES);
+
+        //Now we figure out if our files have changed
+        File main = new File("cache", "main_file_cache.dat2");
+        File xtea = cache.getXTEA().getFile();
+        //Quick way to check if the file has changed. This does not check the idx files, which may cause issues, but the .dat file and the xtea file are
+        //key to allowing/disallowing files from the map cache
+        if (main.lastModified() != cacheCfg.getLong("modified." + main.getName()) || xtea.lastModified() != cacheCfg.getLong("modified." + xtea.getName())) {
+            Log.debug("Cache change detected. Recalculating!");
+            //So we must scan through all of the map files, attempt to parse them, and blacklist broken ones
+            for (int x = 0; x < 256; x++) {
+                for (int y = 0; y < 256; y++) {
+                    Reference ref;
+                    try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://titan.maxgamer.org:9000/coding_rules#rule_key=squid%3AS2293)